src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs

```diff
@@ -278,5 +278,141 @@ public void ResolveSdkTiming ()
 			Assert.AreEqual (validateJavaVersion.MinimumRequiredJdkVersion, "1.8", "MinimumRequiredJdkVersion should be 1.8");
 			Directory.Delete (Path.Combine (Root, path), recursive: true);
 		}
+
+		static object [] TargetFrameworkPairingParameters = new [] {
+			//We support 28, but only 27 is installed
+			new object [] {
+				"Older API Installed", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				null,   //userSelected
+				"27",   //androidApiLevel
+				"27",   //androidApiLevelName
+				"v8.1", //targetFrameworkVersion
+				"28",   //supportedApiLevel
+			},
+			//28 is installed but we only support 27
+			new object [] {
+				"Newer API Installed", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				null,   //userSelected
```

Comment:
Minor quibble (no changes required): I'm finding it hard to mentally map `userSelected` and `supportedApiLevel`/etc.

It might be clearer if these were renamed?

* `userSelected` -> `inputTargetFrameworkVersion`
* `supportedApiLevel` -> `outputSupportedApiLevel`
* etc.

Reply:
Oops, I missed this comment. If I end up needing to make another change here, I'll rename this.

I originally named it `userSelected` to match the variable here: https://github.com/xamarin/xamarin-android/blob/62cddb8c6d134b41eec41c24046df3a6007e9d8f/src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs#L284-L290

But I don't like the name in either place.